src/main/java/org/sonar/plugins/stash/StashIssueReportingPostJob.java

```diff
@@ -129,39 +124,35 @@ private void updateStashWithSonarInfo(StashClient stashClient, StashCredentials
   * Second part of the code necessary for the updateStashWithSonarInfo() method
   *   and third part of the executeOn() method (call of a call) -- squid:MethodCyclomaticComplexity
   */
-  private void postInfoAndPRsActions(PullRequestRef pr, SonarQubeIssuesReport issueReport, int issueThreshold,
-                                       StashDiffReport diffReport, CoverageIssuesReport coverageReport,
-                                       StashClient stashClient, SonarQubeClient sonarQubeClient) {
+  private void postInfoAndPRsActions(PullRequestRef pr, List<Issue> issueReport, int issueThreshold,
+                                       StashDiffReport diffReport,
+                                       StashClient stashClient) {
 
     // Some local definitions
     boolean canApprovePullrequest = config.canApprovePullRequest();
 
-    int issueNumber    = issueReport.countIssues() + coverageReport.countLoweredIssues();
-    int issueTotal     = issueReport.countIssues();
-
-    String sonarQubeURL = sonarQubeClient.getBaseUrl();
+    int issueTotal     = issueReport.size();
 
     // if threshold exceeded, do not push issue list to Stash
-    if (issueNumber >= issueThreshold) {
+    if (issueTotal >= issueThreshold) {
       LOGGER.warn("Too many issues detected ({}/{}): Issues cannot be displayed in Diff view",
                                                      issueTotal, issueThreshold);
     } else {
       // publish SonarQube issue and code coverage
-      stashRequestFacade.postSonarQubeReport(pr, sonarQubeURL, issueReport, diffReport, stashClient);
-      stashRequestFacade.postCoverageReport(pr, sonarQubeURL, coverageReport, diffReport, stashClient);
+      stashRequestFacade.postSonarQubeReport(pr, issueReport, diffReport, stashClient);
     }
 
     if (config.includeAnalysisOverview()) {
-      stashRequestFacade.postAnalysisOverview(pr, sonarQubeURL, issueThreshold, issueReport, coverageReport, stashClient);
+      stashRequestFacade.postAnalysisOverview(pr, issueThreshold, issueReport, stashClient);
     }
 
     // if no new issues and coverage is improved,
     // plugin approves the pull-request
-    if (canApprovePullrequest && (issueNumber == 0) && (coverageReport.getEvolution() >= 0)){
+    if (canApprovePullrequest && (issueTotal == 0)){

```

Comment:
It's clear enough. It cannot be read other way. Sonar will blame you :)

https://docs.oracle.com/javase/tutorial/java/nutsandbolts/operators.html